ts/Core/Dynamics.ts

```diff
@@ -1667,6 +1669,26 @@ extend(Series.prototype, /** @lends Series.prototype */ {
     setName: function (this: Highcharts.Series, name: string): void {
         this.name = this.options.name = this.userOptions.name = name;
         this.chart.isDirtyLegend = true;
+    },
+    /**
+     * Check if the option has changed.
+     *
+     * @private
+     * @function Highcharts.Series#hasOptionChanged
+     *
+     * @param {string} option
+     *
+     * @return {boolean}
+     */
+    hasOptionChanged(this: Highcharts.Series, optionName: string): boolean {
+        const chart = this.chart,
+            plotOptions = chart.options.plotOptions;
+
+        return (this.options as any)[optionName] !== pick(
```

Comment:
Simple `!==` works only for primitives. For objects (e.g. `dataGrouping`) won't work: https://jsfiddle.net/BlackLabel/397zvwm5/4/

Reply:
Use case: https://jsfiddle.net/BlackLabel/397zvwm5/6/ - should it return false?